### ISSUE TITLE
Watch without pulling the images?

### ISSUE THREAD

**Issue by radarsh** [containrrr/watchtower#201]
I don't know if it's been previously asked or addressed. I would like to know if it's possible for watchtower to find out if an image has changed on the repository without having to download it, possibly using the registry API.

In its current state, watchtower keeps incrementing the pull counts on Docker Hub (which may not be such a big deal) but a more serious problem could be that it may exhaust the bandwidth on a limited bandwidth hosting set up if it downloads the entire image every few minutes.

**Comment by lmcnulty4** [comment]
I'd like to know this as well. I don't really know Go but from poking around the source it looks like in the IsContainerStale method first downloads the image and then checks whether to update. Is it not possible to inspect the image without downloading and then download only if it's changed?

**Comment by rbq** [comment]
I'd assume that only metadata (and updated layers) will be transferred? If the image is unchanged, the amount of data transferred should be minimal. Just like the second time you run `docker pull [image]` locally. I actually just tried that and caused 7.4 kB upstream and 29.8 kB downstream traffic. That is like … 320 MB per month and container or so, given a 5 minute interval?

That being said, I landed here because my Docker Hub counter went from 1 to 4000 over a weekend, which is a bit weird. Not sure if it is a Watchtower issue though—maybe instead Docker Hub should not count a pull w/o changes?
